sched/sched/sched_timerexpiration.c

```diff
@@ -424,16 +427,22 @@ static clock_t nxsched_timer_start(clock_t ticks, clock_t interval)
       interval = interval <= (CONFIG_TIMER_ADJUST_USEC / USEC_PER_TICK) ? 0 :
                  interval - (CONFIG_TIMER_ADJUST_USEC / USEC_PER_TICK);
 
-#ifdef CONFIG_SCHED_TICKLESS_ALARM
+#ifdef CONFIG_HRTIMER
+    ret = hrtimer_start(&g_hrtimer_systick_timer,
+                        NSEC_PER_TICK * (ticks + interval),
```

Comment:
as we use HRtimer, but upper half still use tick, we should ensue all the flow to HIGH-Resolution, are we ?
Take a example:
The user call usleep()  ---- HIGH-Resolution
change to ticks in wd_start --- LOW-Resolution
nxsched_timer_start tick    ---- LOW-Resolution
hrtimer_start        ns           ---- HIGH-Resolution
Driver                   tick         ---- LOW-Resolution

Reply:
Nuttx was originally designed as a tick-based framework with a simple and efficient architecture, making it suitable for use on lower-frequency CPUs like the Cortex-M. HRtimers are merely an alternative.

Also, we cannot switch all timer systems to HRtimers, as this would degrade overall system performance. HRtimers are only suitable for high-frequency CPUs with real-time requirements.